test/hexpm/accounts/users_test.exs

```diff
@@ -0,0 +1,237 @@
+        )
+
+      email = Users.get_email("gravatar@example.com")
+      assert email.user_id == organization.user.id
+      assert email.gravatar == true
+
+      # TODO: these audit_logs shall belong to organization or organization user?
+      assert [%{action: "email.gravatar"}, %{action: "email.add"}, _user_update] =
+               Hexpm.Accounts.AuditLogs.all_by(current_user)
+    end
+
+    test "updates gravatar email if it exists" do
+      organization =
+        insert(:organization,
+          user: build(:user, emails: [build(:email, email: "old@example.com", gravatar: true)])
+        )
+
+      current_user = insert(:user)
+
+      {:ok, _updated_user} =
+        Users.update_profile(
+          organization.user,
+          %{"gravatar_email" => "gravatar@example.com"},
+          audit: {current_user, "UA"}
+        )
+
+      email = Users.get_email("gravatar@example.com")
+      assert email.user_id == organization.user.id
+      assert email.gravatar == true
+
+      assert [%{action: "email.gravatar"}, _user_update] =
+               Hexpm.Accounts.AuditLogs.all_by(current_user)
+    end
+
+    test "returns {:error, changeset} when public_email is invalid" do
+      organization = insert(:organization, user: build(:user, emails: []))
+
+      assert {:error, _changeset} =
+               Users.update_profile(
+                 organization.user,
+                 %{"public_email" => "public"},
+                 audit: {build(:user), "UA"}
+               )
+    end
+
+    test "returns {:error, changeset} when gravatar_email is invalid" do
+      organization = insert(:organization, user: build(:user, emails: []))
+
+      assert {:error, _changeset} =
+               Users.update_profile(
+                 organization.user,
+                 %{"gravatar_email" => "gravatar"},
+                 audit: {build(:user), "UA"}
+               )
+    end
+
+    test "removes public email when public_email is empty" do
+      organization =
+        insert(:organization,
+          user: build(:user, emails: [build(:email, email: "old@example.com", public: true)])
+        )
+
+      current_user = insert(:user)
+
+      {:ok, _updated_user} =
+        Users.update_profile(
+          organization.user,
+          %{"public_email" => ""},
+          audit: {current_user, "UA"}
+        )
+
+      assert Users.get_email("old@example.com") == nil
+
+      assert [%{action: "email.remove"}, _user_update] =
+               Hexpm.Accounts.AuditLogs.all_by(current_user)
+    end
+
+    test "does nothing to emails when public_email is empty" do
+      organization = insert(:organization, user: build(:user, emails: []))
+
+      assert {:ok, _updated_user} =
+               Users.update_profile(
+                 organization.user,
+                 %{"public_email" => ""},
+                 audit: {build(:user), "UA"}
+               )
+    end
+
+    test "removes gravatar email when gravatar_email is empty" do
```

Comment:
delete email or set email.gravatar to false?

I chose delete email because org.user emails are not verified anyway, so it's safe to delete them.

Reply:
Sounds perfect.